pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl

```diff
@@ -0,0 +1,329 @@
+
+	>>>
+
+	runtime {
+		docker : "us.gcr.io/tag-team-160914/neovax-tag-rnaseq:v1"
+		disks : "local-disk " + disk_space + " HDD"
+		memory : "64GB"
+		cpu : "8"
+		preemptible: 0
+	}
+
+	output {
+		File aligned_bam = "Aligned.out.bam"
+		File transcriptome_bam = "Aligned.toTranscriptome.out.bam"
+	}
+}
+
+task ExtractUMIs {
+	input {
+		File bam
+		String read1Structure
+		String read2Structure
+	}
+
+	Int disk_space = ceil(2.2 * size(bam, "GB")) + 50
+
+	command <<<
+		fgbio ExtractUmisFromBam --input ~{bam} \
+			--read-structure ~{read1Structure} \
+			--read-structure ~{read2Structure} \
+			--molecular-index-tags RX \
+			--output extractUMIs.out.bam
+	>>>
+
+	runtime {
+		docker : "quay.io/biocontainers/fgbio@sha256:a8e5cf58c318bffba3b2b694a3640ecd9e8106cee2e33b75710c0e8215138b6e"
+		disks : "local-disk " + disk_space + " HDD"
+		preemptible: 0
+	}
+
+	output {
+		File bam_umis_extracted = "extractUMIs.out.bam"
+	}
+}
+
+task rnaseqc2 {
+	input {
+		File bam_file
+		File genes_gtf
+		String sample_id
+		File exon_bed = "gs://gtex-resources/GENCODE/gencode.v26.GRCh38.insert_size_intervals_geq1000bp.bed"
+	}
+	
+	Int disk_space = ceil(size(bam_file, 'GB') + size(genes_gtf, 'GB')) + 100
+
+	command {
+		set -euo pipefail
+		echo $(date +"[%b %d %H:%M:%S] Running RNA-SeQC 2")
+		rnaseqc ~{genes_gtf} ~{bam_file} . -s ~{sample_id} -v --bed ~{exon_bed}
+		echo "  * compressing outputs"
+		gzip *.gct
+		echo $(date +"[%b %d %H:%M:%S] done")
+	}
+
+	output {
+		File gene_tpm = "${sample_id}.gene_tpm.gct.gz"
+		File gene_counts = "${sample_id}.gene_reads.gct.gz"
+		File exon_counts = "${sample_id}.exon_reads.gct.gz"
+		File insert_size_histogram = "${sample_id}.fragmentSizes.txt"
+		File metrics = "${sample_id}.metrics.tsv"
+	}
+
+	runtime {
+		docker: "us.gcr.io/broad-dsde-methods/ckachulis/rnaseqc:2.4.2"
+		memory: "10GB"
+		disks: "local-disk " + disk_space + " HDD"
+		preemptible: 0
+	}
+}
+
+task GetSampleName {
+	input {
+		File bam
+	}
+
+	parameter_meta {
+		bam : {
+			localization_optional : true
+		}
+	}
+
+	command <<<
+		gatk GetSampleName -I ~{bam} -O sample_name.txt
+	>>>
+
+	runtime {
+		docker: "us.gcr.io/broad-gatk/gatk:4.2.0.0"
+		disks: "local-disk 100 HDD"
+	}
+
+	output {
+		String sample_name = read_string("sample_name.txt")
+	}
+}
+
+task CopyReadGroupsToHeader {
+	input {
+		File bam_with_readgroups
+		File bam_without_readgroups
+	}
+
+	Int disk_size = ceil(2.0 * size([bam_with_readgroups, bam_without_readgroups], "GB")) + 10
+	String basename = basename(bam_without_readgroups)
+
+	command <<<
+		samtools view -H ~{bam_without_readgroups} > header.sam
+		samtools view -H ~{bam_with_readgroups} | grep "@RG" >> header.sam
+		samtools reheader header.sam ~{bam_without_readgroups} > ~{basename}
+	>>>
+
+	runtime {
+		docker: "us.gcr.io/broad-dsde-methods/samtools@sha256:0e49b0a5d91c203b8c07f5242277c2060b4b8ea54df8b1d123f990a1ad0588b2"
```

Comment:
Do we know what version of samtools this uses?  Is it okay if we use the most up-to-date?

Reply:
Answering my own question.  This is: Version: 1.11 (using htslib 1.11)

Reply:
Updated to use samtools 1.11 docker.